pkg/volume/glusterfs/glusterfs.go

```diff
@@ -429,14 +440,44 @@ func (d *glusterfsVolumeDeleter) Delete() error {
 		glog.Errorf("glusterfs: failed to create gluster rest client")
 		return fmt.Errorf("glusterfs: failed to create gluster rest client, REST server authentication failed")
 	}
+	volumeinfo, err := cli.VolumeInfo(volumeId)
+	if err != nil {
+		glog.Errorf("glusterfs: failed to get volume details")
+	}
 	err = cli.VolumeDelete(volumeId)
 	if err != nil {
-		glog.V(4).Infof("glusterfs: error when deleting the volume :%s", err)
+		glog.Errorf("glusterfs: error when deleting the volume :%v", err)
 		return err
 	}
 	glog.V(2).Infof("glusterfs: volume %s deleted successfully", volumeName)
-	return nil
 
+	//Deleter takes endpoint and endpointnamespace from pv spec.
+	pvSpec := d.spec.Spec
+	var dynamicEndpoint, dynamicNamespace string
+	if pvSpec.ClaimRef.Namespace != "" {
```

Comment:
@jsafrane I don't think of any empty namespace, but is any special meaning when we get an empty namespace?